src/cli/config.js

```diff
@@ -60,3 +60,26 @@ export const getSchemas = (schemaFilePath: string): [GraphQLSchema, Schema] => {
         return [schemaForValidation, schemaForTypeGeneration];
     }
 };
+
+/**
+ * Find the first item of the `config.generate` array where both:
+ * - no item of `exclude` matches
+ * - at least one item of `match` matches
+ */
+export const findApplicableConfig = (
+    path: string,
+    configs: Array<GenerateConfig> | GenerateConfig,
+): ?GenerateConfig => {
+    if (!Array.isArray(configs)) {
+        configs = [configs];
+    }
+    return configs.find((config) => {
+        if (config.exclude?.some((exclude) => new RegExp(exclude).test(path))) {
+            return false;
+        }
+        if (!config.match) {
```

Comment:
issue: If a general, app-wide config is found before a directory- or file-config, the more specific config will be ignored?

Reply:
Yup! If you have a config with no matchers (and no excludes) it will ignore everything after it.
Fortunately, these configs are all in the same file, so it should be simple to fix.